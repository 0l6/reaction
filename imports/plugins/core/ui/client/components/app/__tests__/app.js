/**
 * Mock Translation component import, as it uses Meteor modules we have a hard time testing with Jest
 */
jest.mock("/imports/plugins/core/ui/client/components", () => ({
  Translation(props) {
      return <span>{props.defaultValue}</span>; // eslint-disable-line
  }
}));

import React from "react";
import { shallow } from "enzyme";
import shallowToJSON from "enzyme-to-json";
import App from "../app";

/**
 * Badge is a display only component
 * It receives props and displays them accordingly
 */

afterEach(() => {
  jest.clearAllMocks();
});

/**
 * Snapshots make sure your UI does not change unexpectedly
 */

test("Basic Rendering: className test", () => {
  const isActionViewOpen = false;
  const hasDashboardAccess = false;
  const currentRoute = {
    route: {
      name: "testingRouteClassName"
    }
  };

  const component = shallow(
    <App
      isActionViewOpen={isActionViewOpen}
      hasDashboardAccess={hasDashboardAccess}
      currentRoute={currentRoute}
    >
      <h1>Testing...</h1>
    </App>
  );
  const tree = shallowToJSON(component);
  expect(tree).toMatchSnapshot();
});
