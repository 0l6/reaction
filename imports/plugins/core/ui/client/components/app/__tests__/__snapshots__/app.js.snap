// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering: className test 1`] = `
<div class="admin page route-testing-route-classnames" id="reactionAppContainer" style="flex: 1 1 auto; height: 100vh;"><h1>Testing...</h1></div>
`;
